### PR TITLE
finished login and signup 

### PR DESCRIPTION
**Pull request by Abigayle-Mercer**
I made a Login/Signup component. This is the first page users are sent to, if they sign up it redirects to the login page, if the login is successful it saves the user id and auth token using secure store and then redirects to the profile page.

### CODE REVIEW

**Comment by PauMinguet** [comment]
Awesome work, also thanks for making it so clear how to implement the auth to my backend calls.

File: frontend/app/_layout.tsx [hunk 1]
```diff
@@ -27,6 +27,7 @@ const StackLayout = () => {
   }
 
   return (
+    // auth provider
```
Comment (AlfredMadere) [inline frontend/app/_layout.tsx:30]: we are no longer using an auth provider, could remove this

File: frontend/app/login.tsx [hunk 2]
```diff
@@ -1,12 +1,233 @@
...
+  const [toggle, setToggle] = useState(false);
+  const [key, onChangeKey] = React.useState("authtoken");
+  const [value, onChangeValue] = React.useState("Your value here");
+
+  const handleSignUp = async () => {
+    //alert("name: " + username + " password: " + password + " email: " + email + " phone: " + phone);
+
+    try {
+      if (
+        name.length > 0 &&
```
Comment (AlfredMadere) [inline frontend/app/login.tsx:34]: In practice you would normally use some kind of form validator like zod. I think for our demo this is fine though

File: frontend/app/login.tsx [hunk 3]
```diff
@@ -1,12 +1,233 @@
...
+          },
+        );
+        //alert("response: " + response.status);
+
+        if (response.status === 200 || response.status === 201) {
+          // Registration successful, redirect to login
+          setSignUp(false);
+          //alert("successfull sign in");
+        } else {
+          throw new Error("Registration failed");
```
Comment (AlfredMadere) [inline frontend/app/login.tsx:60]: if registration fails i'm not sure we should be blowing up the app with an error, probably should show some text that the registration failed and then let them retry. Again, for our demo this is probably fine, hopefully we dont get a network error i guess!

File: frontend/app/login.tsx [hunk 4]
```diff
@@ -1,12 +1,233 @@
...
+            },
+          },
+        );
+
+        if (response.status === 200) {
+          const authToken = response.data.access_token;
+          // Save the auth token
+          save(key, authToken);
+
+          const results = await axios.get(
```
Comment (AlfredMadere) [inline frontend/app/login.tsx:93]: great example for how the others should create their headers, nice

File: frontend/app/login.tsx [hunk 5]
```diff
@@ -1,12 +1,233 @@
...
+          router.push("/profile");
+          alert("success, redirecting to profile right now");
+        } else {
+          throw new Error("Authentication failed");
+        }
+      } else {
+        alert("Please fill out all fields");
+      }
+    } catch (error) {
+      console.error("Login failed:");
```
Comment (AlfredMadere) [inline frontend/app/login.tsx:115]: Generally you would want to throw the error.message here, otherwize theres no real reason to throw the error with a message inside the try block

File: frontend/app/login.tsx [hunk 6]
```diff
@@ -1,12 +1,233 @@
...
+        <Text style={styles.buttonText}>Login</Text>
+      </TouchableOpacity>
+      <TouchableOpacity onPress={() => setSignUp(true)}>
+        <Text style={styles.toggleText}>Dont have an account? Sign Up</Text>
+      </TouchableOpacity>
+    </View>
+  );
+
+  const renderSignUpForm = () => (
+    <View style={styles.container}>
```
Comment (AlfredMadere) [inline frontend/app/login.tsx:148]: Switching between login and signup by state is probably not ideal, adding these to separate paths so you could have a /login route and a /signup route and redirect between them as needed. Not worth changing for demo but worth noting

File: frontend/app/login.tsx [hunk 7]
```diff
@@ -1,12 +1,233 @@
...
+          {
+            headers: {
+              "Content-Type": "application/json",
+            },
+          },
+        );
+        //alert("response: " + response.status);
+
+        if (response.status === 200 || response.status === 201) {
+          // Registration successful, redirect to login
```
Comment (michaelmurrayiv) [inline frontend/app/login.tsx:56]: I like comments describing how the code will respond to the post requests, it really helped me understand what is going on!

**Review by michaelmurrayiv — APPROVED** [review]
I like that you included a lot of error checking and descriptive function names. Good work Abby!

**Review by PauMinguet — APPROVED** [review]
Good work, thank you for making the names so clear for us to later implement auth in our backend calls.

**Comment by Abigayle-Mercer** [comment]
> Good work, thank you for making the names so clear for us to later implement auth in our backend calls.

yep, i knew this was going to affect our entire project so it needed to be simple

File: frontend/app/_layout.tsx (hunk 1 above)
Comment (Abigayle-Mercer) [reply frontend/app/_layout.tsx:30]: sorry forgot to remove that

File: frontend/app/login.tsx (hunk 2 above)
Comment (Abigayle-Mercer) [reply frontend/app/login.tsx:34]: Given the time constraints, I felt that it was better to just get a working login, but thank you in the future i will make  the login requirements stronger

**Pull request merged**